crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -315,4 +316,29 @@ impl<'a> Compressor<'a> {
 
         None
     }
+
+    /// port from [closure-compiler](https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/src/com/google/javascript/jscomp/PeepholeFoldConstants.java#L195)
+    /// void 0 -> void 0
+    /// void 1 -> void 0
+    /// void x -> void 0
+    fn try_reduce_void(&mut self, unary_expr: &UnaryExpression<'a>) -> Option<Expression<'a>> {
+        let can_replace = match &unary_expr.argument {
+            Expression::NumberLiteral(number_literal) => number_literal.value != 0_f64,
+            Expression::Identifier(_) => true,
```

Comment:
It seems like should be checked in [`mayHaveSideEffects`](https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/src/com/google/javascript/jscomp/AstAnalyzer.java#L294-L300). How can we distinguish where the Identifier comes from (left side or not) in our AST? 🤔